### PR TITLE
Add `CWindowsComLifecycle` RAII wrapper for Windows COM library

### PR DESCRIPTION
**Pull request by Robyt3**
I can't reproduce the crash/hang from #5744, so I don't know if this fixes the issue.

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/engine/client/client.cpp [hunk 1]
```diff
@@ -4561,6 +4561,9 @@ int main(int argc, const char **argv)
 {
 #if defined(CONF_PLATFORM_ANDROID)
 	const char **argv = const_cast<const char **>(argv2);
+#endif
+#if defined(CONF_FAMILY_WINDOWS)
```
Comment (def-) [inline src/engine/client/client.cpp:4565]: #elif

**Comment by def-** [comment]
cheeser0613 reports:

> i asked 2 player that having this issue, they say they still can reproduce it

**Comment by Robyt3** [comment]
If they can reproduce it, can they identify the first version where it started happening? Or has it always been like this since we have the feature?

**Comment by def-** [comment]
@Cheeser0613 maybe you should send your findings here directly instead of to me via DM:

> i do ask one of them to try out on 16.1, and nothing happen it just work normally
> i didnt found 16.2.0 patch on official web so i didnt ask them to compare between 16020 and 16022
> to make sure this problem is not only happen on steam version, i also ask them to try out on non-steam 16031, they can reproduce

**Pull request merged**